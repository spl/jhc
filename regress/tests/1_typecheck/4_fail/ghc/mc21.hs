-- Checks that the correct type is used checking the using clause of the group

{-# OPTIONS_GHC -XMonadComprehensions -XTransformListComp #-}

module ShouldFail where
import GHC.Exts( the )

data Unorderable = Gnorf | Pinky | Brain

foo = [ length x
      | x <- [Gnorf, Brain]
      , then group using take 5
      ]
