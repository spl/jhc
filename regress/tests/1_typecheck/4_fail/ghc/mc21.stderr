
mc21.hs:12:26:
    Couldn't match type `a' with `[a]'
      `a' is a rigid type variable bound by
          a type expected by the context: [a] -> [[a]] at mc21.hs:12:9
    Expected type: [a] -> [[a]]
      Actual type: [a] -> [a]
    In the return type of a call of `take'
    In the expression: take 5
